**Integrantes:** Marvin Francisconi | Ricardo Illanes  
**Curso:** Visualización de Datos (ADY1104)  
**Caso:** StreamView Analytics

## Indice:

* [1. Problema de negocio](#sec-1)
* [2. Audiencia objetivo y próposito comunicacional](#sec-2)
* [3. Descripción e integración de las fuentes de datos](#sec-3)
* [4. EDA](#sec-4)
    * [4.1 Descripción del EDA](#sec-4-1)

### &rarr; Carga de datasets

In [5]:
import pandas as pd

# Cargar los datasets
usuarios = pd.read_csv("StreamView_Analytics_CSV/usuarios.csv")
contenidos = pd.read_csv("StreamView_Analytics_CSV/contenidos.csv")
suscripciones = pd.read_csv("StreamView_Analytics_CSV/suscripciones.csv")
dispositivos = pd.read_csv("StreamView_Analytics_CSV/dispositivos.csv")
reproducciones = pd.read_csv("StreamView_Analytics_CSV/reproducciones.csv")
calificaciones = pd.read_csv("StreamView_Analytics_CSV/calificaciones.csv")
interacciones = pd.read_csv("StreamView_Analytics_CSV/interacciones.csv")


# Diccionario para trabajar con todos los datasets
datasets = {
    "Usuarios": usuarios,
    "Contenidos": contenidos,
    "Suscripciones": suscripciones,
    "Dispositivos": dispositivos,
    "Reproducciones": reproducciones,
    "Calificaciones": calificaciones,
    "Interacciones": interacciones
}


# Exploración inicial
for nombre, df in datasets.items():
    print("=" * 70)
    print(nombre)
    print("=" * 70)
    
    print(f"Filas: {df.shape[0]}")
    print(f"Columnas: {df.shape[1]}")
    
    print("\nColumnas:")
    print(df.columns.tolist())
    
    print("\nValores nulos:")
    print(df.isnull().sum())
    
    print("\nPrimeras 5 filas:")
    display(df.head())
    
    print()

Usuarios
Filas: 1500
Columnas: 8

Columnas:
['usuario_id', 'fecha_registro', 'pais', 'ciudad', 'edad', 'segmento_edad', 'genero', 'canal_adquisicion']

Valores nulos:
usuario_id           0
fecha_registro       0
pais                 0
ciudad               0
edad                 0
segmento_edad        0
genero               0
canal_adquisicion    0
dtype: int64

Primeras 5 filas:


,usuario_id,fecha_registro,pais,ciudad,edad,segmento_edad,genero,canal_adquisicion
0,U00001,2024-02-06,Chile,Concepción,33,25-34,Prefiere no indicar,Orgánico
1,U00002,2025-05-13,Chile,Valparaíso,63,55+,Femenino,Redes sociales
2,U00003,2023-03-17,México,Ciudad de México,32,25-34,Femenino,Referido
3,U00004,2023-01-25,España,Madrid,47,45-54,Femenino,Orgánico
4,U00005,2025-03-16,Chile,La Serena,41,35-44,Femenino,Orgánico



Contenidos
Filas: 420
Columnas: 14

Columnas:
['contenido_id', 'titulo', 'tipo_contenido', 'genero_principal', 'subgenero', 'clasificacion_edad', 'anio_estreno', 'duracion_min', 'temporadas', 'episodios', 'idioma_original', 'pais_origen', 'exclusivo', 'fecha_alta_catalogo']

Valores nulos:
contenido_id           0
titulo                 0
tipo_contenido         0
genero_principal       0
subgenero              0
clasificacion_edad     0
anio_estreno           0
duracion_min           0
temporadas             0
episodios              0
idioma_original        0
pais_origen            0
exclusivo              0
fecha_alta_catalogo    0
dtype: int64

Primeras 5 filas:


,contenido_id,titulo,tipo_contenido,genero_principal,subgenero,clasificacion_edad,anio_estreno,duracion_min,temporadas,episodios,idioma_original,pais_origen,exclusivo,fecha_alta_catalogo
0,C0001,Marea Alto 1,Serie,Comedia,Romántica,16+,2012,49,1,9,Inglés,Estados Unidos,No,2023-08-19
1,C0002,Tiempo Azul 2,Película,Documental,Sociedad,16+,2001,131,0,1,Español,España,No,2025-07-21
2,C0003,Eco Global 3,Película,Acción,Supervivencia,18+,2005,123,0,1,Inglés,Estados Unidos,Sí,2025-05-09
3,C0004,Destino Rojo 4,Película,Aventura,Exploración,13+,1998,98,0,1,Inglés,España,No,2025-04-08
4,C0005,Eco Norte 5,Documental,Drama,Histórico,13+,2000,73,0,1,Español,España,No,2023-11-24



Suscripciones
Filas: 1500
Columnas: 9

Columnas:
['suscripcion_id', 'usuario_id', 'plan', 'fecha_inicio', 'fecha_fin', 'estado', 'precio_mensual_usd', 'renovacion_automatica', 'motivo_cancelacion']

Valores nulos:
suscripcion_id              0
usuario_id                  0
plan                        0
fecha_inicio                0
fecha_fin                1085
estado                      0
precio_mensual_usd          0
renovacion_automatica       0
motivo_cancelacion       1085
dtype: int64

Primeras 5 filas:


,suscripcion_id,usuario_id,plan,fecha_inicio,fecha_fin,estado,precio_mensual_usd,renovacion_automatica,motivo_cancelacion
0,S00001,U00001,Premium,2024-02-06,NaN,Activa,13.99,Sí,NaN
1,S00002,U00002,Básico,2025-05-13,NaN,Activa,6.99,Sí,NaN
2,S00003,U00003,Básico,2023-03-17,2025-12-20,Cancelada,6.99,No,Precio
3,S00004,U00004,Estándar,2023-01-25,NaN,Activa,9.99,Sí,NaN
4,S00005,U00005,Estándar,2025-03-16,NaN,Activa,9.99,Sí,NaN



Dispositivos
Filas: 2492
Columnas: 6

Columnas:
['dispositivo_id', 'usuario_id', 'tipo_dispositivo', 'sistema_operativo', 'version_app', 'fecha_registro_dispositivo']

Valores nulos:
dispositivo_id                0
usuario_id                    0
tipo_dispositivo              0
sistema_operativo             0
version_app                   0
fecha_registro_dispositivo    0
dtype: int64

Primeras 5 filas:


,dispositivo_id,usuario_id,tipo_dispositivo,sistema_operativo,version_app,fecha_registro_dispositivo
0,D00001,U00001,Móvil,iOS,7.2,2024-09-25
1,D00002,U00002,Smart TV,Tizen,7.2,2025-10-17
2,D00003,U00003,Consola,PlayStation,7.1,2025-02-09
3,D00004,U00004,Smart TV,webOS,7.2,2025-09-17
4,D00005,U00005,Smart TV,Roku OS,7.0,2025-08-29



Reproducciones
Filas: 25000
Columnas: 11

Columnas:
['reproduccion_id', 'usuario_id', 'contenido_id', 'dispositivo_id', 'fecha_hora_inicio', 'tipo_dispositivo', 'calidad_video', 'minutos_reproducidos', 'porcentaje_completado', 'abandono_temprano', 'buffering_segundos']

Valores nulos:
reproduccion_id          0
usuario_id               0
contenido_id             0
dispositivo_id           0
fecha_hora_inicio        0
tipo_dispositivo         0
calidad_video            0
minutos_reproducidos     0
porcentaje_completado    0
abandono_temprano        0
buffering_segundos       0
dtype: int64

Primeras 5 filas:


,reproduccion_id,usuario_id,contenido_id,dispositivo_id,fecha_hora_inicio,tipo_dispositivo,calidad_video,minutos_reproducidos,porcentaje_completado,abandono_temprano,buffering_segundos
0,R000001,U00975,C0223,D01616,2025-07-02 01:02:46,Tablet,HD,61.5,67.6,No,5.6
1,R000002,U00234,C0019,D00385,2025-09-16 17:48:26,Computador,4K,65.4,64.1,No,7.8
2,R000003,U00475,C0132,D00785,2025-02-21 20:29:07,Tablet,HD,28.2,53.3,No,20.8
3,R000004,U01392,C0123,D02316,2025-08-24 13:24:47,Móvil,HD,11.6,20.4,No,4.7
4,R000005,U00582,C0105,D00964,2025-06-27 18:09:27,Computador,SD,27.2,73.6,No,0.9



Calificaciones
Filas: 3119
Columnas: 7

Columnas:
['calificacion_id', 'usuario_id', 'contenido_id', 'fecha_calificacion', 'puntuacion_1_5', 'recomendaria', 'categoria_comentario']

Valores nulos:
calificacion_id         0
usuario_id              0
contenido_id            0
fecha_calificacion      0
puntuacion_1_5          0
recomendaria            0
categoria_comentario    0
dtype: int64

Primeras 5 filas:


,calificacion_id,usuario_id,contenido_id,fecha_calificacion,puntuacion_1_5,recomendaria,categoria_comentario
0,K00001,U00975,C0363,2025-08-22 08:20:32,4,Sí,Contenido
1,K00002,U00975,C0135,2025-10-26 13:24:35,5,Sí,Historia
2,K00003,U00975,C0324,2025-04-22 20:21:42,4,Sí,Contenido
3,K00004,U00975,C0299,2025-06-07 00:40:30,4,Sí,Experiencia general
4,K00005,U00234,C0008,2025-12-23 20:16:42,4,Sí,Actuación



Interacciones
Filas: 7438
Columnas: 7

Columnas:
['interaccion_id', 'usuario_id', 'contenido_id', 'fecha_hora', 'tipo_interaccion', 'canal', 'origen_interfaz']

Valores nulos:
interaccion_id      0
usuario_id          0
contenido_id        0
fecha_hora          0
tipo_interaccion    0
canal               0
origen_interfaz     0
dtype: int64

Primeras 5 filas:


,interaccion_id,usuario_id,contenido_id,fecha_hora,tipo_interaccion,canal,origen_interfaz
0,I000001,U00975,C0324,2025-04-22 20:26:42,Clic recomendación,Web,Recomendados
1,I000002,U00975,C0299,2025-06-04 01:52:30,Compartir,Web,Búsqueda
2,I000003,U00975,C0382,2025-10-18 20:44:19,Clic recomendación,App,Inicio
3,I000004,U00975,C0363,2025-08-19 09:22:32,Me gusta,App,Inicio
4,I000005,U00975,C0135,2025-10-19 13:50:35,No me gusta,Web,Recomendados


### Se van a tomar 6 características que se consideran importantes para poder generar un problema de negocio:  
* `Usuario`: País
* `Suscripcines`: Estado
* `Reproducciones`: Dispositivo
* `Calidad`: Calidad de vídeo
* `Contenidos`: Tipo
* `Género`: Género

In [8]:
print("USUARIOS")
print(usuarios["pais"].value_counts())
print("\nSegmento de edad:")
print(usuarios["segmento_edad"].value_counts())

print("\nSUSCRIPCIONES")
print(suscripciones["plan"].value_counts())
print("\nEstado:")
print(suscripciones["estado"].value_counts())

print("\nREPRODUCCIONES")
print("\nDispositivos:")
print(reproducciones["tipo_dispositivo"].value_counts())

print("\nCalidad:")
print(reproducciones["calidad_video"].value_counts())

print("\nCONTENIDOS")
print("\nTipo:")
print(contenidos["tipo_contenido"].value_counts())

print("\nGénero:")
print(contenidos["genero_principal"].value_counts())

USUARIOS
pais
Chile        470
México       267
Perú         192
Colombia     174
Argentina    171
España       159
Ecuador       67
Name: count, dtype: int64

Segmento de edad:
segmento_edad
35-44    442
25-34    438
18-24    338
45-54    217
55+       65
Name: count, dtype: int64

SUSCRIPCIONES
plan
Estándar    653
Básico      555
Premium     292
Name: count, dtype: int64

Estado:
estado
Activa       1085
Cancelada     415
Name: count, dtype: int64

REPRODUCCIONES

Dispositivos:
tipo_dispositivo
Móvil         7641
Smart TV      7010
Computador    6290
Tablet        3183
Consola        876
Name: count, dtype: int64

Calidad:
calidad_video
Full HD    10394
HD          8861
4K          3802
SD          1943
Name: count, dtype: int64

CONTENIDOS

Tipo:
tipo_contenido
Película      198
Serie         167
Documental     55
Name: count, dtype: int64

Género:
genero_principal
Drama              70
Comedia            57
Acción             47
Thriller           43
Ciencia ficción    40
Crimen  

### Resumen corto:  

* 1.500 usuarios, principalmente de Chile, México, Perú, Colombia, Argentina y España.
* La mayor concentración está entre 25 y 44 años.
* 1.085 suscripciones activas y 415 canceladas, aproximadamente 27,7% canceladas.
* El plan más utilizado es Estándar.
* El consumo está concentrado en Móvil, Smart TV y Computador.
* La calidad más utilizada es Full HD, seguida de HD.
* El catálogo tiene principalmente Películas y Series.
* Los géneros más numerosos del catálogo son Drama, Comedia y Acción.

<a id = "sec-1"></a>
# 1. Problema de negocio  

StreamView Analytics necesita comprender los patrones asociados a la permanencia y cancelación de suscripciones, considerando las características de sus usuarios y su comportamiento de consumo, para apoyar decisiones orientadas a la retención y experiencia del cliente.

## Pregunta de análisis:
&rarr; **¿Qué patrones de usuarios, suscripciones y consumo se relacionan con la permanencia o cancelación de las suscripciones?**

<a id="sec-2"></a>
# 2. Audiencia objetivo y propósito comunicacional.

## Audiencia objetivo:  
Equipo de gestión y toma de decisiones de StreamView Analytics.

## Propósito educacional:  
Comunicar de manera clara los principales patrones relacionados con la permanencia y cancelación de suscripciones, utilizando información sobre usuarios, suscripciones y comportamiento de consumo para facilitar la toma de decisiones.

<a id="sec-3"></a>
# 3. Descripción e integración de las fuentes de datos.  

La descripción principal se encuentra en archivo README.txt, pincha [aquí](StreamView_Analytics_CSV/README.txt) para acceder.  

Se utilizaron siete fuentes de datos en formato CSV correspondientes a usuarios, contenidos, suscripciones, dispositivos, reproducciones, calificaciones e interacciones. Las fuentes fueron integradas mediante identificadores comunes como `usuario_id`, `contenido_id` y `dispositivo_id`, permitiendo relacionar las características de los usuarios con sus suscripciones, comportamiento de consumo, contenidos e interacciones.  

&rarr; ***Esta integración permite analizar el comportamiento de los usuarios desde distintas dimensiones y estudiar su relación con la permanencia o cancelación de las suscripciones.***

<a id="sec-4"></a>
# 4. EDA

---
# Exploración de suscripciones
---

In [9]:
estado_suscripciones = (
    suscripciones["estado"]
    .value_counts()
    .to_frame("cantidad")
)

estado_suscripciones["porcentaje"] = (
    estado_suscripciones["cantidad"]
    / estado_suscripciones["cantidad"].sum()
    * 100
).round(2)

display(estado_suscripciones)

,cantidad,porcentaje
estado,,
Activa,1085,72.33
Cancelada,415,27.67


In [10]:
cancelacion_plan = pd.crosstab(
    suscripciones["plan"],
    suscripciones["estado"]
)

cancelacion_plan["total"] = cancelacion_plan.sum(axis=1)

cancelacion_plan["tasa_cancelacion"] = (
    cancelacion_plan["Cancelada"]
    / cancelacion_plan["total"]
    * 100
).round(2)

display(cancelacion_plan)

estado,Activa,Cancelada,total,tasa_cancelacion
plan,,,,
Básico,310,245,555,44.14
Estándar,518,135,653,20.67
Premium,257,35,292,11.99


In [11]:
cancelacion_pais = pd.crosstab(
    usuarios["pais"],
    suscripciones["estado"]
)

cancelacion_pais["total"] = cancelacion_pais.sum(axis=1)

cancelacion_pais["tasa_cancelacion"] = (
    cancelacion_pais["Cancelada"]
    / cancelacion_pais["total"]
    * 100
).round(2)

display(
    cancelacion_pais.sort_values(
        "tasa_cancelacion",
        ascending=False
    )
)

estado,Activa,Cancelada,total,tasa_cancelacion
pais,,,,
Ecuador,43,24,67,35.82
Colombia,120,54,174,31.03
Perú,133,59,192,30.73
España,112,47,159,29.56
México,193,74,267,27.72
Argentina,129,42,171,24.56
Chile,355,115,470,24.47


In [12]:
cancelacion_edad = pd.crosstab(
    usuarios["segmento_edad"],
    suscripciones["estado"]
)

cancelacion_edad["total"] = cancelacion_edad.sum(axis=1)

cancelacion_edad["tasa_cancelacion"] = (
    cancelacion_edad["Cancelada"]
    / cancelacion_edad["total"]
    * 100
).round(2)

display(cancelacion_edad)

estado,Activa,Cancelada,total,tasa_cancelacion
segmento_edad,,,,
18-24,225,113,338,33.43
25-34,325,113,438,25.80
35-44,320,122,442,27.60
45-54,171,46,217,21.20
55+,44,21,65,32.31


In [13]:
cancelacion_canal = pd.crosstab(
    usuarios["canal_adquisicion"],
    suscripciones["estado"]
)

cancelacion_canal["total"] = cancelacion_canal.sum(axis=1)

cancelacion_canal["tasa_cancelacion"] = (
    cancelacion_canal["Cancelada"]
    / cancelacion_canal["total"]
    * 100
).round(2)

display(
    cancelacion_canal.sort_values(
        "tasa_cancelacion",
        ascending=False
    )
)

estado,Activa,Cancelada,total,tasa_cancelacion
canal_adquisicion,,,,
Publicidad digital,243,108,351,30.77
Orgánico,305,125,430,29.07
Redes sociales,253,96,349,27.51
Alianza comercial,86,30,116,25.86
Referido,198,56,254,22.05


---
# Consumo por estado de suscripción 
---

In [14]:
reproducciones_suscripcion = reproducciones.merge(
    suscripciones[["usuario_id", "estado", "plan"]],
    on="usuario_id",
    how="left"
)

display(reproducciones_suscripcion.head())

,reproduccion_id,usuario_id,contenido_id,dispositivo_id,fecha_hora_inicio,tipo_dispositivo,calidad_video,minutos_reproducidos,porcentaje_completado,abandono_temprano,buffering_segundos,estado,plan
0,R000001,U00975,C0223,D01616,2025-07-02 01:02:46,Tablet,HD,61.5,67.6,No,5.6,Cancelada,Estándar
1,R000002,U00234,C0019,D00385,2025-09-16 17:48:26,Computador,4K,65.4,64.1,No,7.8,Activa,Premium
2,R000003,U00475,C0132,D00785,2025-02-21 20:29:07,Tablet,HD,28.2,53.3,No,20.8,Cancelada,Premium
3,R000004,U01392,C0123,D02316,2025-08-24 13:24:47,Móvil,HD,11.6,20.4,No,4.7,Activa,Estándar
4,R000005,U00582,C0105,D00964,2025-06-27 18:09:27,Computador,SD,27.2,73.6,No,0.9,Activa,Básico


In [16]:
consumo_estado = (
    reproducciones_suscripcion
    .groupby("estado")
    .agg(
        reproducciones=("reproduccion_id", "count"),
        minutos_totales=("minutos_reproducidos", "sum"),
        minutos_promedio=("minutos_reproducidos", "mean"),
        completado_promedio=("porcentaje_completado", "mean")
    )
    .round(2)
)

display(consumo_estado)

,reproducciones,minutos_totales,minutos_promedio,completado_promedio
estado,,,,
Activa,20118,1148201.5,57.07,71.52
Cancelada,4882,275289.3,56.39,71.58


In [17]:
reproducciones_por_usuario = (
    reproducciones_suscripcion
    .groupby(["estado", "usuario_id"])
    .size()
    .groupby(level=0)
    .mean()
    .round(2)
)

display(reproducciones_por_usuario)

estado
Activa       18.54
Cancelada    11.76
dtype: float64

In [19]:
minutos_por_usuario = (
    reproducciones_suscripcion
    .groupby(["estado", "usuario_id"])["minutos_reproducidos"]
    .sum()
    .groupby(level=0)
    .mean()
    .round(2)
)

display(minutos_por_usuario)

estado
Activa       1058.25
Cancelada     663.35
Name: minutos_reproducidos, dtype: float64

In [20]:
completado_por_estado = (
    reproducciones_suscripcion
    .groupby("estado")["porcentaje_completado"]
    .mean()
    .round(2)
)

display(completado_por_estado)

estado
Activa       71.52
Cancelada    71.58
Name: porcentaje_completado, dtype: float64

In [21]:
abandono_estado = pd.crosstab(
    reproducciones_suscripcion["estado"],
    reproducciones_suscripcion["abandono_temprano"],
    normalize="index"
).mul(100).round(2)

display(abandono_estado)

abandono_temprano,No,Sí
estado,,
Activa,99.58,0.42
Cancelada,99.71,0.29


---
# Consumo de contenido según sus tipos
---

In [22]:
consumo_contenido = reproducciones.merge(
    contenidos,
    on="contenido_id",
    how="left"
)

display(
    consumo_contenido[
        [
            "reproduccion_id",
            "contenido_id",
            "titulo",
            "tipo_contenido",
            "genero_principal",
            "minutos_reproducidos",
            "porcentaje_completado"
        ]
    ].head()
)

,reproduccion_id,contenido_id,titulo,tipo_contenido,genero_principal,minutos_reproducidos,porcentaje_completado
0,R000001,C0223,Instante Nocturno 223,Película,Animación,61.5,67.6
1,R000002,C0019,Última Final 19,Película,Thriller,65.4,64.1
2,R000003,C0132,Secreto Bajo 132,Serie,Ciencia ficción,28.2,53.3
3,R000004,C0123,Vértigo Perdido 123,Serie,Comedia,11.6,20.4
4,R000005,C0105,Memoria Alto 105,Serie,Drama,27.2,73.6


In [23]:
consumo_tipo = (
    consumo_contenido
    .groupby("tipo_contenido")
    .agg(
        reproducciones=("reproduccion_id", "count"),
        minutos_totales=("minutos_reproducidos", "sum"),
        completado_promedio=("porcentaje_completado", "mean")
    )
    .round(2)
    .sort_values("reproducciones", ascending=False)
)

display(consumo_tipo)

,reproducciones,minutos_totales,completado_promedio
tipo_contenido,,,
Película,11499,917069.3,68.70
Serie,10192,317398.1,73.19
Documental,3309,189023.4,76.27


In [24]:
consumo_genero = (
    consumo_contenido
    .groupby("genero_principal")
    .agg(
        reproducciones=("reproduccion_id", "count"),
        minutos_totales=("minutos_reproducidos", "sum"),
        completado_promedio=("porcentaje_completado", "mean")
    )
    .round(2)
    .sort_values("reproducciones", ascending=False)
)

display(consumo_genero)

,reproducciones,minutos_totales,completado_promedio
genero_principal,,,
Drama,4552,260208.0,73.86
Comedia,3946,205720.6,69.85
Acción,3312,181312.5,71.02
Thriller,2626,156566.6,73.16
Ciencia ficción,2469,146408.9,70.87
Aventura,1929,112472.0,70.43
Crimen,1892,119854.5,69.27
Romance,1890,97694.1,71.92
Animación,1333,73880.5,71.07


In [25]:
top_contenidos = (
    consumo_contenido
    .groupby(["contenido_id", "titulo"])
    .agg(
        reproducciones=("reproduccion_id", "count"),
        minutos_totales=("minutos_reproducidos", "sum"),
        completado_promedio=("porcentaje_completado", "mean")
    )
    .round(2)
    .sort_values("reproducciones", ascending=False)
    .head(10)
)

display(top_contenidos)

,,reproducciones,minutos_totales,completado_promedio
contenido_id,titulo,,,
C0304,Pulso Libre 304,99,2513.3,74.66
C0032,Pulso Global 32,93,7501.5,65.05
C0098,Secreto Nocturno 98,91,1996.6,73.13
C0055,Punto Bajo 55,90,2547.2,70.75
C0194,Punto Oculto 194,89,3409.6,78.18
C0377,Ciudad Final 377,88,6463.3,71.31
C0068,Vértigo Humano 68,87,6352.5,72.29
C0249,Luz Imposible 249,86,3333.8,76.02
C0243,Noche Norte 243,85,7535.1,67.67


In [27]:
consumo_completo = reproducciones.merge(
    suscripciones[["usuario_id", "estado"]],
    on="usuario_id",
    how="left"
).merge(
    contenidos[
        [
            "contenido_id",
            "tipo_contenido",
            "genero_principal"
        ]
    ],
    on="contenido_id",
    how="left"
)

In [28]:
genero_estado = pd.crosstab(
    consumo_completo["genero_principal"],
    consumo_completo["estado"],
    normalize="columns"
).mul(100).round(2)

display(
    genero_estado.sort_values(
        "Activa",
        ascending=False
    )
)

estado,Activa,Cancelada
genero_principal,,
Drama,18.16,18.39
Comedia,15.56,16.71
Acción,13.23,13.33
Thriller,10.47,10.63
Ciencia ficción,9.79,10.22
Aventura,7.79,7.39
Romance,7.64,7.21
Crimen,7.58,7.54
Animación,5.45,4.85


---
# Consumo según sus dispositivos
---

In [29]:
dispositivo_estado = pd.crosstab(
    reproducciones_suscripcion["tipo_dispositivo"],
    reproducciones_suscripcion["estado"],
    normalize="columns"
).mul(100).round(2)

display(dispositivo_estado)

estado,Activa,Cancelada
tipo_dispositivo,,
Computador,24.83,26.51
Consola,3.22,4.69
Móvil,30.49,30.89
Smart TV,28.47,26.26
Tablet,12.99,11.66


In [30]:
calidad_estado = pd.crosstab(
    reproducciones_suscripcion["calidad_video"],
    reproducciones_suscripcion["estado"],
    normalize="columns"
).mul(100).round(2)

display(calidad_estado)

estado,Activa,Cancelada
calidad_video,,
4K,16.15,11.31
Full HD,42.46,37.94
HD,34.16,40.72
SD,7.22,10.04


In [31]:
buffering_estado = (
    reproducciones_suscripcion
    .groupby("estado")["buffering_segundos"]
    .agg(
        promedio="mean",
        mediana="median"
    )
    .round(2)
)

display(buffering_estado)

,promedio,mediana
estado,,
Activa,8.26,7.4
Cancelada,8.24,7.4


In [32]:
experiencia_estado = (
    reproducciones_suscripcion
    .groupby("estado")
    .agg(
        reproducciones=("reproduccion_id", "count"),
        minutos_promedio=("minutos_reproducidos", "mean"),
        completado_promedio=("porcentaje_completado", "mean"),
        buffering_promedio=("buffering_segundos", "mean")
    )
    .round(2)
)

display(experiencia_estado)

,reproducciones,minutos_promedio,completado_promedio,buffering_promedio
estado,,,,
Activa,20118,57.07,71.52,8.26
Cancelada,4882,56.39,71.58,8.24


<a id="sec-4-1"></a>
## 4.1 Descripción del EDA
Se realizó un análisis exploratorio de las principales variables relacionadas con usuarios, suscripciones, consumo de contenidos y experiencia de reproducción. El objetivo fue identificar diferencias y patrones asociados al estado de las suscripciones.

En primer lugar, se observó que el **72,33% de las suscripciones se encuentran activas y el 27,67% canceladas**. Al analizar la cancelación según el plan contratado, el plan Básico presenta la mayor tasa de cancelación, con **44,14%**, seguido por Estándar con **20,67%** y Premium con **11,99%**.

Respecto al comportamiento de consumo, los usuarios con suscripción activa presentan un mayor nivel promedio de actividad: registran aproximadamente **58% más reproducciones por usuario y 60% más minutos de reproducción por usuario** que los usuarios cuya suscripción figura como cancelada. Sin embargo, el porcentaje promedio de contenido completado es prácticamente igual entre ambos grupos (**71,52% frente a 71,58%**).

También se analizaron las preferencias de contenido. Las distribuciones de géneros presentan patrones similares entre usuarios activos y cancelados, por lo que el género de contenido no muestra diferencias relevantes entre ambos grupos. Las películas concentran la mayor cantidad de reproducciones, seguidas por las series y los documentales.

Finalmente, se evaluó la experiencia técnica de reproducción. La distribución de dispositivos es similar entre ambos grupos y el **buffering promedio es prácticamente idéntico** (8,26 segundos en usuarios activos y 8,24 segundos en cancelados). En cuanto a calidad de video, los usuarios activos presentan una mayor proporción de reproducciones en 4K y Full HD, mientras que los cancelados presentan una mayor proporción en HD y SD.

Estos resultados permiten orientar el análisis hacia la identificación de los factores y patrones más asociados a la permanencia o cancelación de las suscripciones, los cuales serán profundizados mediante las visualizaciones y el dashboard.
